catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/TemplateTransformer.java

```diff
@@ -45,20 +52,67 @@
 public class TemplateTransformer {
   private static final Logger LOGGER = LoggerFactory.getLogger(TemplateTransformer.class);
 
+  private final FilterWriter writer;
+
+  public TemplateTransformer(FilterWriter writer) {
+    this.writer = writer;
+  }
+
   public static boolean invalidFormTemplate(Metacard metacard) {
-    return new TemplateTransformer().toFormTemplate(metacard) == null;
+    return TemplateTransformer.toFormTemplate(metacard) == null;
+  }
+
+  /** Convert the JSON representation of a FormTemplate to a QueryTemplateMetacard. */
+  @Nullable
+  public Metacard toQueryTemplateMetacard(Map<String, Object> formTemplate) {
+    Map<String, Object> filterJson = (Map) formTemplate.get("filterTemplate");
+    String title = (String) formTemplate.get("title");
+    String description = (String) formTemplate.get("description");
+
+    TransformVisitor<JAXBElement> visitor = new TransformVisitor<>(new XmlModelBuilder());
+    try {
+      VisitableJsonElementImpl.create(new FilterNodeMapImpl(filterJson)).accept(visitor);
+      JAXBElement filter = visitor.getResult();
+      if (!filter.getDeclaredType().equals(FilterType.class)) {
+        LOGGER.error(
+            "Error occurred during filter processing, root type should be a {} but was {}",
+            FilterType.class.getName(),
+            filter.getDeclaredType().getName());
+        return null;
+      }
+
+      String id = (String) formTemplate.get("id");
+      QueryTemplateMetacard metacard =
+          (id == null)
+              ? new QueryTemplateMetacard(title, description)
+              : new QueryTemplateMetacard(title, description, id);
+
+      String filterXml = writer.marshal(filter);
+      metacard.setFormsFilter(filterXml);
+      Map<String, Object> querySettings = (Map<String, Object>) formTemplate.get("querySettings");
+      if (querySettings != null) {
+        metacard.setQuerySettings(querySettings);
+      }
+      return metacard;
+    } catch (JAXBException e) {
+      LOGGER.error("XML generation failed for query template metacard's filter", e);
```

Comment:
❓ Should these really be at `error`? is there something that an administrator can do for these?  Seems like `debug` would be appropriate. 
❓ If we're failing to generate the xml, then returning null and silently ignoring the null -- is that okay if the users data is silently dropped with no feedback to them? (meaning we are doing best effort here as opposed to fail all -- is that the intended response?)